modules/library/main/src/main/java/org/geotools/data/AbstractDataStoreFactory.java

```diff
@@ -30,8 +30,10 @@
 import org.opengis.parameter.ParameterDescriptorGroup;
 import org.opengis.parameter.ParameterValue;
 
-
+@Deprecated
 /**
+ * DEPRECATED - Use org.geotools.data.store.ContentDataStore instead
```

Comment:
If you want to provide a message with advice about what to use instead use the javadocs @deprecated tag - it will include your message in the compiler warnings.

```
@deprecated Use org.geotools.data.store.ContentDataStore instead
```